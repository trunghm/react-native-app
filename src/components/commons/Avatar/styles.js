import EStyleSheet from "react-native-extended-stylesheet";
import GlobalStore from "../../../themes/GlobalStore";

const styles = () => EStyleSheet.create({
  container: {
    width : 45,
    height: 45,
    borderRadius:45,
    borderWidth:0.5,
    alignItems:"center",
    justifyContent:"center",
    backgroundColor: GlobalStore.color.avatarBg,

  },

  text:{
    color : GlobalStore.color.primary
  },
  circle:{
    width : 45,
    height: 45,
    borderRadius:45,
    borderWidth:0.5,
  }

});

export default styles;
