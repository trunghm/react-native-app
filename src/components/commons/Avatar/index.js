import React, {Component} from 'react';
import styles from "./styles";
import {
  Text,
  View,
  Image
} from 'react-native';

export default function Avatar({url, text}) {
  return (
    <View style={styles().container}>
      <Image style={styles().circle}/>
      <Text style={styles().text}>{text}</Text>
    </View>);
}



