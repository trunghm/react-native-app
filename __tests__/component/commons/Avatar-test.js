/**
 * @format
 */

import 'react-native';
import React from 'react';
import Avatar from '../../../src/components/commons/Avatar';

// Note: test renderer must be required after react-native.
import {mount, shallow} from 'enzyme';
import renderer from  "react-test-renderer";
import EStyleSheet from 'react-native-extended-stylesheet';


const originalError = console.error;

beforeAll(() => {
  console.error = jest.fn();
  EStyleSheet.build({
  });
});

afterAll(() => {
  console.error = originalError;
});

describe('Testing  Avatar component', () => {

  it('renders snapshot as expected', () => {
    const wrapper = renderer.create(
      <Avatar text={"H"}/>
    );

    expect(wrapper).toMatchSnapshot();

  });

  it('renders as expected', () => {
    const wrapper = mount(
      <Avatar text={"H"}/>
    );

    expect(wrapper).toMatchSnapshot();

  });

});
