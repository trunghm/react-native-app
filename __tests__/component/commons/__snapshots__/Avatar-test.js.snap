// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing  Avatar component renders as expected 1`] = `ReactWrapper {}`;

exports[`Testing  Avatar component renders snapshot as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#d9e2e7",
      "borderRadius": 45,
      "borderWidth": 0.5,
      "height": 45,
      "justifyContent": "center",
      "width": 45,
    }
  }
>
  <Image />
  <Text
    style={
      Object {
        "color": "#1496bb",
      }
    }
  >
    H
  </Text>
</View>
`;
